src/panels/scene-viewer/SceneViewer.tsx

```diff
@@ -25,43 +30,52 @@ import { getUrlTempVarName, undecorateName } from 'common/variables';
 
 export const SceneViewer = (props: SceneViewerPropsFromParent) => {
   const styles = getStyles(props.width, props.height);
-  const selectedNodeRef = useRef<string>();
+  const id = useMemo(() => uuid(), []);
+  const { getSceneNodeByRef, getSelectedSceneNodeRef } = useSceneComposerApi(id);
 
   const { search } = useLocation();
 
-  const onTargetObjectChanged = useCallback(
-    (objectData: TargetObjectData) => {
-      const anchorData = objectData.data;
+  const onWidgetClick = useCallback(
+    (objectData: IWidgetClickEvent) => {
+      const anchorData =
+        objectData.additionalComponentData?.[
+          objectData.componentTypes.findIndex((type) => type === KnownComponentType.Tag)
+        ];
 
-      if (anchorData?.eventType === 'click') {
-        selectedNodeRef.current = anchorData.anchorNodeRef;
+      if (anchorData) {
         const targetLink = getValidHttpUrl(anchorData.navLink);
         if (targetLink) {
           window.open(targetLink.toString());
         }
-      } else if (anchorData?.eventType === 'change') {
-        if (anchorData.isSelected) {
-          selectedNodeRef.current = anchorData.anchorNodeRef;
-          const dashboardId = anchorData.navLink?.params?.[MERGE_DASHBOARD_TARGET_ID_KEY];
-          mergeDashboard(dashboardId).then((options) => {
-            updateUrlParams(
-              options?.customSelEntityVarName || props.options.customSelEntityVarName,
-              options?.customSelCompVarName || props.options.customSelCompVarName,
-              options?.customSelPropertyVarName || props.options.customSelPropertyVarName,
-              anchorData
-            );
-          });
-        } else {
-          if (selectedNodeRef.current === anchorData.anchorNodeRef) {
-            selectedNodeRef.current = undefined;
-            updateUrlParams(
-              props.options.customSelEntityVarName,
-              props.options.customSelCompVarName,
-              props.options.customSelPropertyVarName,
-              anchorData
-            );
-          }
-        }
+      }
+    },
+    [props.options.customSelEntityVarName, props.options.customSelCompVarName, props.options.customSelPropertyVarName]
```

Comment:
Don't need these deps since they're not used in the `useCallback`. Can remove the `useCallback` altogether.

Reply:
done